crates/storage-query-datafusion/src/log/schema.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(dead_code)]
+
+use datafusion::arrow::datatypes::DataType;
+
+use crate::table_macro::*;
+
+define_table!(
+    log(
+    /// Log ID
+    id: DataType::UInt32,
+
+    /// Segment index
+    segment_index: DataType::UInt32,
+
+    /// Segment start lsn
+    base_lsn: DataType::UInt64,
+
+    /// Segment provider kind
+    kind: DataType::Utf8,
+
+    // replicated loglet specific params
+
+    /// Loglet ID
+    loglet_id: DataType::UInt64,
+
+    // todo: should this be merged into
+    // a utf8 `N1:1` for example?
+
+    /// Node id of the node running the sequencer
+    seq_node_id: DataType::UInt32,
+
+    /// Node generation of the node running the sequencer
+    seq_node_gen: DataType::UInt32,
+
+    /// loglet replication factor
+    replication: DataType::Utf8,
+
+    /// Loglet node set
+    nodeset: DataType::Utf8,
```

Comment:
This right now is a list of nodes formatted as string `[N1, N2]` which is not helpful in queries, joins, etc... Instead I think it's better if we:
- Completely flatten it (we get a row per log-server), but will create way more redundant data
- Create a `cluster_log_server` table which can be queried with log_id + segment_index.